src/Infrastructure/Persistence/Doctrine/Repository/Regulation/LocationRepository.php

```diff
@@ -67,19 +64,7 @@ public function findAllForMapAsGeoJSON(
             : ($temporaryOnly
                 ? 'AND ro.end_date IS NOT NULL'
                 : '');
-
-        $upcomingOnly = $includeUpcomingRegulations && !$includePastRegulations;
-        $pastOnly = !$includeUpcomingRegulations && $includePastRegulations;
-        $presentOnly = !$includeUpcomingRegulations && !$includePastRegulations;
-
-        $regulationDatesWhereClause =
-            $upcomingOnly
-            ? 'AND (((ro.end_date >= :now OR ro.end_date IS NULL) AND ro.start_date <= :now) OR ro.start_date > :now)'
-            : ($pastOnly
-                ? 'AND (((ro.end_date >= :now OR ro.end_date IS NULL) AND ro.start_date <= :now) OR ro.end_date < :now)'
-                : ($presentOnly
-                    ? 'AND ((ro.end_date >= :now OR ro.end_date IS NULL) AND ro.start_date <= :now)'
-                    : ''));
+        $regulationTypeCondition = $measureTypes ? \sprintf('AND m.type IN (%s)', "'" . implode("', '", $measureTypes) . "'") : '';
```

Comment:
Attention ! En faisant ce type d'interpolation on introduit un risque [d'injection SQL](https://www.reddit.com/r/explainlikeimfive/comments/4p5r9c/eli5_how_does_sql_injection_work_and_how_can_you/)

L'injection SQL consiste pour un attaquant à injecter du SQL de son choix via un paramètre de saisie utilisateur, ce qui lui permet de faire des opérations malveillantes sur la DB, jusqu'à potentiellement supprimer toute la DB ou faire d'autres dégâts

Ici en plus je ne pense pas qu'on ait de validation particulière sur le contenu de l'array `measureTypes`

Donc on pourrait forcer, via l'URL, la saisie d'une chaîne comme `'; DROP DATABASE dialog;` et hop tout disparaît

La solution "standard" à ce problème consiste à "échapper" les saisies utilisateur pour empêcher les caractères spéciaux comme `'` ou `;`, etc. C'est fait automatiquement par Doctrine quand on utilise des paramètres nommés type `:monParametre` configurés avec `->setParameters()`.

Donc ici il faut plutôt faire :

```suggestion
        $regulationTypeCondition = $measureTypes ? 'AND m.type IN (:measureTypes)' : '';
```

Et ajouter `measureTypes` aux paramètres.

Concrètement pour des requêtes complexes comme ça j'aime bien initialiser un array `$parameters`, le remplir avec les paramètres dans les cas où il le faut, puis appeler `->setParameters($parameters)`.

Tu as un exemple de ça dans `findAllRegulations()` du `RegulationOrderRecordRepository`.

Ici ça donnerait quelque chose comme ça :

```php
$parameters = [
    'status' => RegulationOrderRecordStatusEnum::PUBLISHED->value,
];

if ($measureTypes) {
    $regulationTypeCondition = 'AND m.type IN (:measureTypes)';
    $parameters['measureTypes'] = $measureTypes;
} else {
    $regulationTypeCondition = '';
}

// ...

$rows = $this->getEntityManager()->getConnection()->fetchAllAssociative(
    ...,
    $parameters
);

Reply:
Concernant les parapètres nommés j'y ai pensé mais malheureusement  m.type IN (:measureTypes)' : ''; ça ne fonctionne pas ...